### PR TITLE
Release 2.12.0

### PR DESCRIPTION
**Pull request by svenklemm**
This release contains performance improvements for compressed hypertables
and continuous aggregates and bug fixes since the 2.11.2 release.
We recommend that you upgrade at the next available opportunity.

This release moves all internal functions from the _timescaleb_internal
schema into the _timescaledb_functions schema. This separates code from
internal data objects and improves security by allowing more restrictive
permissions for the code schema. If you are calling any of those internal
functions you should adjust your code as soon as possible. This version
also includes a compatibility layer that allows calling them in the old
location but that layer will be removed in 2.14.0.

**PostgreSQL 12 support removal announcement**
Following the deprecation announcement for PostgreSQL 12 in TimescaleDB 2.10,
PostgreSQL 12 is not supported starting with TimescaleDB 2.12.
Currently supported PostgreSQL major versions are 13, 14 and 15.
PostgreSQL 16 support will be added with a following TimescaleDB release.

**Features**
* https://github.com/timescale/timescaledb/pull/5137 Insert into index during chunk compression
* https://github.com/timescale/timescaledb/pull/5150 MERGE support on hypertables
* https://github.com/timescale/timescaledb/pull/5515 Make hypertables support replica identity
* https://github.com/timescale/timescaledb/pull/5586 Index scan support during UPDATE/DELETE on compressed hypertables
* https://github.com/timescale/timescaledb/pull/5596 Support for partial aggregations at chunk level
* https://github.com/timescale/timescaledb/pull/5599 Enable ChunkAppend for partially compressed chunks
* https://github.com/timescale/timescaledb/pull/5655 Improve the number of parallel workers for decompression
* https://github.com/timescale/timescaledb/pull/5758 Enable altering job schedule type through `alter_job`
* https://github.com/timescale/timescaledb/pull/5805 Make logrepl markers for (partial) decompressions
* https://github.com/timescale/timescaledb/pull/5809 Relax invalidation threshold table-level lock to row-level when refreshing a Continuous Aggregate
* https://github.com/timescale/timescaledb/pull/5839 Support CAgg names in chunk_detailed_size
* https://github.com/timescale/timescaledb/pull/5852 Make set_chunk_time_interval CAggs aware
* https://github.com/timescale/timescaledb/pull/5868 Allow ALTER TABLE ... REPLICA IDENTITY (FULL|INDEX) on materialized hypertables (continuous aggregates)
* https://github.com/timescale/timescaledb/pull/5875 Add job exit status and runtime to log
* https://github.com/timescale/timescaledb/pull/5909 CREATE INDEX ONLY ON hypertable creates index on chunks

**Bugfixes**
* https://github.com/timescale/timescaledb/pull/5860 Fix interval calculation for hierarchical CAggs
* https://github.com/timescale/timescaledb/pull/5894 Check unique indexes when enabling compression
* https://github.com/timescale/timescaledb/pull/5951 _timescaledb_internal.create_compressed_chunk doesn't account for existing uncompressed rows
* https://github.com/timescale/timescaledb/pull/5988 Move functions to _timescaledb_functions schema
* https://github.com/timescale/timescaledb/pull/5788 Chunk_create must add an existing table or fail
* https://github.com/timescale/timescaledb/pull/5872 Fix duplicates on partially compressed chunk reads
* https://github.com/timescale/timescaledb/pull/5918 Fix crash in COPY from program returning error
* https://github.com/timescale/timescaledb/pull/5990 Place data in first/last function in correct mctx
* https://github.com/timescale/timescaledb/pull/5991 Call eq_func correctly in time_bucket_gapfill
* https://github.com/timescale/timescaledb/pull/6015 Correct row count in EXPLAIN ANALYZE INSERT .. ON CONFLICT output
* https://github.com/timescale/timescaledb/pull/6035 Fix server crash on UPDATE of compressed chunk
* https://github.com/timescale/timescaledb/pull/6044 Fix server crash when using duplicate segmentby column
* https://github.com/timescale/timescaledb/pull/6045 Fix segfault in set_integer_now_func
* https://github.com/timescale/timescaledb/pull/6053 Fix approximate_row_count for CAggs
* https://github.com/timescale/timescaledb/pull/6081 Improve compressed DML datatype handling
* https://github.com/timescale/timescaledb/pull/6084 Propagate parameter changes to decompress child nodes

**Thanks**
* @ajcanterbury for reporting a problem with lateral joins on compressed chunks
* @alexanderlaw for reporting multiple server crashes
* @lukaskirner for reporting a bug with monthly continuous aggregates
* @mrksngl for reporting a bug with unusual user names
* @willsbit for reporting a crash in time_bucket_gapfill

Disable-check: force-changelog-file
Disable-check: commit-count

### CODE REVIEW

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:24]:
```suggestion
* #5586 Index scan support during UPDATE/DELETE on compressed hypertables
```

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:9]: Initial suggestion on mentioning performance improvements for compression and caggs

```suggestion
This release contains performance improvements for compressed hypertables and continuous aggregates
and bug fixes since the 2.11.2 release.
```

@svenklemm @fabriziomello @antekresic do you think that we could elevate this by adding a quick list of noteworthy features or should we leave this to that?

File: CHANGELOG.md
Comment (jnidzwetzki) [inline CHANGELOG.md:42]: Nit: 
```suggestion
* #5788 Chunk_create must add an existing table or fail
```

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:18]: Should we stab this for 2.14 or is it too early to do so?

```suggestion
location but that layer will be removed in 2.14.
```

**Review by iroussos — COMMENTED** [review]
Thank you @svenklemm, I have added a few small comments

**Review by jnidzwetzki — APPROVED** [review]

**Comment by iroussos** [comment]
I don't see https://github.com/timescale/timescaledb/pull/6067 included - the updates required for Data Tiering GA is a major reason why we are having 2.12 a little bit earlier, so we should make sure that everything is included before tagging.

@svenklemm do you mind also making sure with @zilder, @gayyappan that this is the only one remaining and that there is nothing else we forgot to add to the 2.12 milestone?

File: CHANGELOG.md
Comment (fabriziomello) [reply CHANGELOG.md:9]: Leave it as is IMHO

**Comment by svenklemm** [comment]
> I don't see #6067 included - the updates required for Data Tiering GA is a major reason why we are having 2.12 a little bit earlier, so we should make sure that everything is included before tagging.
> 
> @svenklemm do you mind also making sure with @zilder, @gayyappan that this is the only one remaining and that there is nothing else we forgot to add to the 2.12 milestone?

#6067 has not been merged yet, will add it once its merged.

**Comment by horzsolt** [comment]
Shouldn't we include PG12 deprecation? @iroussos, what would be a good one-liner for the release notes?

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:20]:
```suggestion

**PostgreSQL 12 support removal announcement**
Following the deprecation announcement for PostgreSQL 12 in TimescaleDB 2.10,
PostgreSQL 12 is not supported starting with TimescaleDB 2.12.
Please upgrade your PostgreSQL server to a newer PostgreSQL major version to 
be able to upgrade to TimescaleDB 2.12.
Currently supported PostgreSQL major versions: 13, 14, 15
PostgreSQL 16 support will be added with a following TimescaleDB release.
```

Comment (iroussos) [reply CHANGELOG.md:20]: What do you think @svenklemm? Maybe too wordy? I thought of adding the last 2 lines to be safe

Comment (svenklemm) [reply CHANGELOG.md:20]: Maybe cut out the middle sentence but other than that looks good to me

File: CHANGELOG.md
Comment (konskov) [inline CHANGELOG.md:42]: nit: 
```suggestion
* #5868 Allow ALTER TABLE ... REPLICA IDENTITY (FULL|INDEX) on materialized hypertables (continuous aggregates)
```

**Review by konskov — APPROVED** [review]

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:25]:
```suggestion
```

**Pull request merged**
